lib/elastic_record/percolator_model.rb

```diff
@@ -34,6 +38,26 @@ def percolate(document)
 
         elastic_search.filter(query).limit(5000)
       end
+
+      private
+
+        def check_mapped_doctype_name
+          mapped_doctypes = elastic_connection.json_get("/#{base_class.name.demodulize.underscore.pluralize}/_mapping").values.first['mappings']
+
+          if mapped_doctypes.keys.count > 1
+            mapped_percolator_doctype(mapped_doctypes)
+          end
```

Comment:
Why not use `multiple_doctypes?`

Reply:
Because I'd have to reference `elastic_index` which I found creates an infinite loop. Index initializes which calls `doctype` on all models passed which again calls `check_mapped_doctype_name` which would call `elastic_index` again.

I realize this is all kind of messy, still a WIP. I just wanted to see if this solution would actually work before cleaning it up.